src/community/backup-restore/core/src/main/java/org/geoserver/backuprestore/BackupRestoreItem.java

```diff
@@ -0,0 +1,219 @@
+import org.geotools.filter.text.cql2.CQLException;
+import org.geotools.filter.text.ecql.ECQL;
+import org.opengis.filter.Filter;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.annotation.BeforeStep;
+import org.springframework.util.Assert;
+
+import com.thoughtworks.xstream.XStream;
+
+/**
+ * @author Alessio Fabiani, GeoSolutions S.A.S.
+ *
+ */
+public abstract class BackupRestoreItem<T> {
+
+    protected Backup backupFacade;
+
+    private Catalog catalog;
+
+    protected XStreamPersister xstream;
+
+    private XStream xp;
+
+    private boolean isNew;
+
+    private AbstractExecutionAdapter currentJobExecution;
+
+    private boolean dryRun;
+
+    private boolean bestEffort;
+
+    private XStreamPersisterFactory xStreamPersisterFactory;
+
+    private Filter filter;
+
+    public BackupRestoreItem(Backup backupFacade, XStreamPersisterFactory xStreamPersisterFactory) {
+        this.backupFacade = backupFacade;
+        this.xStreamPersisterFactory = xStreamPersisterFactory;
+    }
+
+    /**
+     * @return the xStreamPersisterFactory
+     */
+    public XStreamPersisterFactory getxStreamPersisterFactory() {
+        return xStreamPersisterFactory;
+    }
+
+    /**
+     * @return the xp
+     */
+    public XStream getXp() {
+        return xp;
+    }
+
+    /**
+     * @param xp the xp to set
+     */
+    public void setXp(XStream xp) {
+        this.xp = xp;
+    }
+
+    /**
+     * @return the catalog
+     */
+    public Catalog getCatalog() {
+        return catalog;
+    }
+
+    /**
+     * @return the isNew
+     */
+    public boolean isNew() {
+        return isNew;
+    }
+
+    /**
+     * @return the currentJobExecution
+     */
+    public AbstractExecutionAdapter getCurrentJobExecution() {
+        return currentJobExecution;
+    }
+
+    /**
+     * @return the dryRun
+     */
+    public boolean isDryRun() {
+        return dryRun;
+    }
+
+    /**
+     * @return the bestEffort
+     */
+    public boolean isBestEffort() {
+        return bestEffort;
+    }
+
+    /**
+     * @return the filter
+     */
+    public Filter getFilter() {
+        return filter;
+    }
+
+    /**
+     * @param filter the filter to set
+     */
+    public void setFilter(Filter filter) {
+        this.filter = filter;
+    }
+
+    @BeforeStep
+    public void retrieveInterstepData(StepExecution stepExecution) {
+        // Accordingly to the running execution type (Backup or Restore) we
+        // need to validate resources against the official GeoServer Catalog (Backup)
+        // or the temporary one (Restore).
+        //
+        // For restore operations the order matters.
+        JobExecution jobExecution = stepExecution.getJobExecution();
+        this.xstream = xStreamPersisterFactory.createXMLPersister();
+        if (backupFacade.getRestoreExecutions() != null
```

Comment:
So the code is still going through the facade instead of using the native spring-batch approach: http://stackoverflow.com/questions/8117060/storing-in-jobexecutioncontext-from-tasklet-and-accessing-in-another-tasklet